helix-term/src/ui/mod.rs

```diff
@@ -33,7 +33,26 @@ pub fn prompt(
     completion_fn: impl FnMut(&Editor, &str) -> Vec<prompt::Completion> + 'static,
     callback_fn: impl FnMut(&mut crate::compositor::Context, &str, PromptEvent) + 'static,
 ) {
-    let mut prompt = Prompt::new(prompt, history_register, completion_fn, callback_fn);
+    prompt_with_input(
+        cx,
+        prompt,
+        String::new(),
+        history_register,
+        completion_fn,
+        callback_fn,
+    )
```

Comment:
Is it a good idea to have `prompt` call `prompt_with_input`?

Reply:
how shall we share the common logic? do you recommend extract another function?

Reply:
Hmm, I am not really sure, maybe create another (private) function and call it `prompt_inner`?

Reply:
done, please take another look, thanks!